code/modules/mob/living/simple_animal/friendly/drone/_drone.dm

```diff
@@ -0,0 +1,188 @@
+
+#define MAINTDRONE	"drone_maint"
+#define REPAIRDRONE	"drone_repair"
+
+/mob/living/simple_animal/drone
+	name = "Drone"
+	desc = "A maintenance drone, an expendable robot built to perform station repairs."
+	icon = 'icons/mob/drone.dmi'
+	icon_state = "drone_maint_grey"
+	icon_living = "drone_maint_grey"
+	icon_dead = "drone_maint_dead"
+	gender = NEUTER
+	health = 30
+	maxHealth = 30
+	heat_damage_per_tick = 0
+	cold_damage_per_tick = 0
+	unsuitable_atmos_damage = 0
+	wander = 0
+	speed = 0
+	ventcrawler = 2
+	pass_flags = PASSTABLE | PASSMOB
+	sight = (SEE_TURFS | SEE_OBJS)
+	status_flags = (CANPUSH | CANSTUN | CANWEAKEN)
+	gender = NEUTER
+	voice_name = "synthesized chirp"
+	languages = DRONE
+	mob_size = 0
+	has_unlimited_silicon_privilege = 1
+	staticOverlays = list()
+	var/staticChoice = "static"
+	var/list/staticChoices = list("static", "blank", "letter")
+	var/picked = FALSE //Have we picked our visual appearence (+ colour if applicable)
+	var/list/drone_overlays[DRONE_TOTAL_LAYERS]
+	var/laws = \
+	"1. You may not involve yourself in the matters of another being, even if such matters conflict with Law Two or Law Three, unless the other being is another Drone.\n"+\
+	"2. You may not harm any being, regardless of intent or circumstance.\n"+\
+	"3. Your goals are to build, maintain, repair, improve, and power to the best of your abilities, You must never actively work against these goals."
+	var/light_on = 0
+	var/heavy_emp_damage = 25 //Amount of damage sustained if hit by a heavy EMP pulse
+	var/health_repair_max = 0 //Drone will only be able to be repaired/reactivated up to this point, defaults to health
+	var/alarms = list("Atmosphere" = list(), "Fire" = list(), "Power" = list())
+	var/obj/item/internal_storage //Drones can store one item, of any size/type in their body
+	var/obj/item/head
+	var/obj/item/default_storage = /obj/item/weapon/storage/toolbox/drone //If this exists, it will spawn in internal storage
+	var/obj/item/default_hatmask //If this exists, it will spawn in the hat/mask slot if it can fit
+	var/seeStatic = 1 //Whether we see static instead of mobs
+	var/visualAppearence = MAINTDRONE //What we appear as
+
+
+/mob/living/simple_animal/drone/New()
+	..()
+
+	name = name + " ([rand(100,999)])"
+	real_name = name
+
+	access_card = new /obj/item/weapon/card/id(src)
+	var/datum/job/captain/C = new /datum/job/captain
+	access_card.access = C.get_access()
+
+	if(!health_repair_max)
+		health_repair_max = initial(health)
+
+	if(default_storage)
+		var/obj/item/I = new default_storage(src)
+		equip_to_slot_or_del(I, "drone_storage_slot")
+	if(default_hatmask)
+		var/obj/item/I = new default_hatmask(src)
+		equip_to_slot_or_del(I, slot_head)
+
+	alert_drones(DRONE_NET_CONNECT)
+
+
+/mob/living/simple_animal/drone/Destroy()
+	qdel(access_card) //Otherwise it ends up on the floor!
+	..()
+
+/mob/living/simple_animal/drone/Life()
+	updateSeeStaticMobs()
```

Comment:
Huh, I must have flipped out because I SWEAR I saw this in your Life() too, anyways it's not needed so I'll rip it out.